backend/coins/eth/blockbook/blockbook.go

```diff
@@ -0,0 +1,163 @@
+// Copyright 2025 Shift Crypto AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package blockbook
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"math/big"
+	"net/http"
+	"net/url"
+	"path"
+
+	"github.com/BitBoxSwiss/bitbox-wallet-app/backend/accounts"
+	"github.com/BitBoxSwiss/bitbox-wallet-app/backend/coins/eth/erc20"
+	"github.com/BitBoxSwiss/bitbox-wallet-app/backend/coins/eth/rpcclient"
+	ethtypes "github.com/BitBoxSwiss/bitbox-wallet-app/backend/coins/eth/types"
+	"github.com/BitBoxSwiss/bitbox-wallet-app/util/errp"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"golang.org/x/time/rate"
+)
+
+// callsPerSec is the number of blockbook requests allowed
+// per second.
+// TODO bznein determine a better value for this.
+const callsPerSec = 3.8
+
+// Blockbook is a rate-limited blockbook api client.
+type Blockbook struct {
+	url        string
+	httpClient *http.Client
+	limiter    *rate.Limiter
+}
+
+// NewBlockbook creates a new instance of EtherScan.
+func NewBlockbook(chainId string, httpClient *http.Client) *Blockbook {
+	// TODO chainID is not used here, do we have blockbook running for SEPETH as well?
+	return &Blockbook{
+		url:        "https://bb1.shiftcrypto.io/api/",
+		httpClient: httpClient,
+		limiter:    rate.NewLimiter(rate.Limit(callsPerSec), 1),
+	}
+}
+
+func (blockbook *Blockbook) call(ctx context.Context, handler string, params url.Values, result interface{}) error {
+	if err := blockbook.limiter.Wait(ctx); err != nil {
+		return errp.WithStack(err)
+	}
+
+	reqUrl := blockbook.url + handler + "?" + params.Encode()
+
+	response, err := blockbook.httpClient.Get(reqUrl)
+	if err != nil {
+		return errp.WithStack(err)
+	}
+	defer func() { _ = response.Body.Close() }()
+	if response.StatusCode != http.StatusOK {
+		return errp.Newf("expected 200 OK, got %d", response.StatusCode)
+	}
+	body, err := io.ReadAll(response.Body)
+	if err != nil {
+		return errp.WithStack(err)
+	}
+	if err := json.Unmarshal(body, result); err != nil {
+		return errp.Newf("unexpected response from blockbook: %s", string(body))
+	}
+
+	return nil
+}
+
+func (blockbook *Blockbook) address(ctx context.Context, account common.Address, result interface{}) error {
+	params := url.Values{}
+	address := account.Hex()
+
+	if err := blockbook.call(ctx, path.Join("address", address), params, result); err != nil {
```

Comment:
This is a URL not a filepath, so on Windows this would use a backslash?

Reply:
Yeah good catch, I moved away from path for other reasons and forgot about this one!